### PR TITLE
node/os: Make scopeid of type undefined in NetworkInterfaceInfoIPv4

### PR DESCRIPTION
**Pull request by phm07**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://nodejs.org/api/os.html#osnetworkinterfaces>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

When `NetworkInterfaceInfoIPv4` doesn't include `scopeid: undefined`, `scopeid` will not exist on union type `NetworkInterfaceInfo`. See [this GitHub issue](https://github.com/microsoft/TypeScript/issues/50925).

### CODE REVIEW

File: types/node/os.d.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ declare module 'os' {
     }
     interface NetworkInterfaceInfoIPv4 extends NetworkInterfaceBase {
         family: 'IPv4';
+        scopeid: undefined;
```
Comment (peterblazejewicz) [inline types/node/os.d.ts:31]:
```suggestion
        scopeid?: undefined;
```
there is no single convention, that one would be aligned with most of the other entires and with support for strict optional property types

File: types/node/ts4.8/os.d.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ declare module 'os' {
     }
     interface NetworkInterfaceInfoIPv4 extends NetworkInterfaceBase {
         family: 'IPv4';
+        scopeid: undefined;
```
Comment (peterblazejewicz) [inline types/node/ts4.8/os.d.ts:31]:
```suggestion
        scopeid?: undefined;
```

File: types/node/v14/os.d.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ declare module 'os' {
 
     interface NetworkInterfaceInfoIPv4 extends NetworkInterfaceBase {
         family: "IPv4";
+        scopeid: undefined;
```
Comment (peterblazejewicz) [inline types/node/v14/os.d.ts:24]:
```suggestion
        scopeid?: undefined;
```

File: types/node/v14/ts4.8/os.d.ts [hunk 4]
```diff
@@ -21,6 +21,7 @@ declare module 'os' {
 
     interface NetworkInterfaceInfoIPv4 extends NetworkInterfaceBase {
         family: "IPv4";
+        scopeid: undefined;
```
Comment (peterblazejewicz) [inline types/node/v14/ts4.8/os.d.ts:24]:
```suggestion
        scopeid?: undefined;
```

File: types/node/v16/ts4.8/os.d.ts [hunk 5]
```diff
@@ -28,6 +28,7 @@ declare module 'os' {
     }
     interface NetworkInterfaceInfoIPv4 extends NetworkInterfaceBase {
         family: 'IPv4';
+        scopeid: undefined;
```
Comment (peterblazejewicz) [inline types/node/v16/ts4.8/os.d.ts:31]:
```suggestion
        scopeid?: undefined;
```

**Comment by peterblazejewicz** [comment]
@pauhull please update from master, that would fix IC errors

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@pauhull thanks!

**Comment by pauhull** [comment]
Ready to merge

**Pull request merged**